### PR TITLE
fix: incorrect status display of DevOps Projects

### LINKED ISSUE #2591: Incorrect status display of DevOps Projects
**Describe the bug**

Incorrect status display of DevOps Projects. Please see the screenshot:

image

**Versions used(KubeSphere/Kubernetes)**
KubeSphere: v3.2.0
Kubernetes: v1.21.5

**To Reproduce**
1. Stop Deployment `devops-jenkins`
2. Create a DevOps Project
3. See DevOps Projects

**Expected behavior**

When the DevOps Project is pending to synchronize, the status should be `Pending` instead of `Active`.

/area devops
/kind bug
/priority high

### PR DESCRIPTION
**Pull request by mangoGoForward**
Signed-off-by: mango <xu.weiKyrie@foxmail.com>

### What type of PR is this?

/kind bug

### Which issue(s) this PR fixes:

Fixes #2591

### Special notes for reviewers:
image

### Does this PR introduced a user-facing change?

```release-note
None
```

### CODE REVIEW

File: src/utils/object.mapper.js [hunk 1]
```diff
@@ -1128,7 +1131,9 @@ const DevOpsMapper = item => {
     devops: get(item, 'metadata.name'),
     workspace: get(item, 'metadata.labels["kubesphere.io/workspace"]'),
     namespace: get(item, 'status.adminNamespace'),
-    status: deletionTimestamp ? 'Terminating' : phase || 'Active',
+    status: deletionTimestamp
+      ? 'Terminating'
+      : phase || get(item, syncStatusKey) || 'Active',
```
Comment (JohnNiang) [inline src/utils/object.mapper.js:1136]: Everything is okay, but I suggest making string style of `get(item, syncStatusKey)` be title, in order to keep the same style with `Terminating` and `Active`.

**Review by JohnNiang — COMMENTED** [review]
Hi @mangoGoForward , I've left a comment below, please take a look at it.

File: src/utils/object.mapper.js (hunk 1 above)
Comment (mangoGoForward) [reply src/utils/object.mapper.js]: Done.

Comment (JohnNiang) [reply src/utils/object.mapper.js]: In my manual test, I got `pending` status instead of `Pending`.

image

Comment (mangoGoForward) [reply src/utils/object.mapper.js]: Yes, It determined by the value of `devopsproject.devops.kubesphere.io/syncstatus`
https://github.com/kubesphere/console/blob/b9345f9f21ca2648c78ae88febc5f759f3c5d66e/locales/en/base.js#L321-L322

Comment (LinuxSuRen) [reply src/utils/object.mapper.js]: In this case, I prefer to use `Pending`.

Comment (JohnNiang) [reply src/utils/object.mapper.js]: In fact, value of `devopsproject.devops.kubesphere.io/syncstatus` has the following values:

```go
const (
	StatusActive     = "active"
	StatusDeleted    = "deleted"
	StatusDeleting   = "deleting"
	StatusFailed     = "failed"
	StatusPending    = "pending"
	StatusWorking    = "working"
	StatusSuccessful = "successful"
)
```
> https://github.com/kubesphere/ks-devops/blob/c96f7386e7cc716e7d2d9269c7376cd56921f07a/pkg/models/devops/common.go#L50-L56

I'd prefer to capitalize them in `object.mapper.js`.

Comment (mangoGoForward) [reply src/utils/object.mapper.js]: Ok, and next need to do:

1. Capitalize the first word.
2. Add lost items to language package.

Is my thinking correct？

Comment (JohnNiang) [reply src/utils/object.mapper.js]: I have no problem with your idea.

Comment (mangoGoForward) [reply src/utils/object.mapper.js]: Done. @JohnNiang @LinuxSuRen

**Comment by JohnNiang** [comment]
Thanks for your contribution!
/lgtm

**Comment by harrisonliu5** [comment]
Maybe you should update your commit according to this rule (https://www.conventionalcommits.org/en/v1.0.0/). @mangoGoForward

**Comment by mangoGoForward** [comment]
> Maybe you should update your commit according to this rule (https://www.conventionalcommits.org/en/v1.0.0/). @mangoGoForward

Thanks, I will be attention in later commits.

**Comment by harrisonliu5** [comment]
Thanks for your contribution!
/approve

**Comment by JohnNiang** [comment]
Hi @mangoGoForward , please wait for `/label tide/merge-method-squash` to work properly before merging this PR. I'm working on the fix.

**Comment by mangoGoForward** [comment]
> Hi @mangoGoForward , please wait for `/label tide/merge-method-squash` to work properly before merging this PR. I'm working on the fix.

Got it.

**Pull request merged**